config/workflow_resource_params_conf.xml.sample

```diff
@@ -0,0 +1,33 @@
+<root>
+	<parameters>
+		<param label="Processors" name="processors" type="integer" size="2" min="1" max="64" value="" help="Number of processing cores, 'ppn' value (1-64). Leave blank to use default value." />
+		<param label="Memory" name="memory" type="integer" size="3" min="1" max="256" value="" help="Memory size in gigabytes, 'pmem' value (1-256). Leave blank to use default value." />
+		<param label="Time" name="time" type="integer" size="3" min="1" max="744" value="" help="Maximum job time in hours, 'walltime' value (1-744). Leave blank to use default value." />
+  		<param label="Project" name="project" type="text" value="" help="Project to assign resource allocation to. Leave blank to use default value." />
+  		<param label="Workflow Job Priority" name="priority" type="select" value="med" help="What priority should the jobs in this workflow run at? (Overrides any declared job priority)">
+			<option value="low" label="Low"/>
+			<option value="med" label="Medium"/>
+			<option value="high" label="High"/>
+			<option value="ultra" label="Ultra"/>
+			<option value="plus_ultra" label="Plus Ultra"/>
+		</param>
+	</parameters>
+	<groups default="default">
```

Comment:
Using Galaxy groups this way should be possible but I would prefer it be an extension point thing and not bake into the XML format. This is how we do job parameters for instance. How about this file just defines the parameters above (so the root tag is ``root``). And then in galaxy.ini you can define a default set of parameters for all workflows -or- a Python method. That Python method will be passed all the stuff it could want - the workflow itself, the user, etc... and should return a list of parameters to display. In a second pass it could generate the parameters on the fly even.

Reply:
Could you elaborate on how/where I would define such a python method?

One of the goals we had for this change was to allow our admin to easily change which users had access to what priority levels. This is why I used the xml to define group access to parameters so we could manage the users(groups) from within galaxy and modify the xml when we need to change what groups access a specific parameter.

I agree with you that baking groups into the xml is probably not the best approach but I am not sure how I should make the changes to allow our intended functionality.